src/utils/read-context.ts

```diff
@@ -0,0 +1,93 @@
+import assert from './assert';
+
+// TODO: rather than define these types partially we might want to read them from octokit similar to what is done in src/commands/merge
+// we'll want the single types (eg review vs reviews)
```

Comment:
I looked *very briefly* at this but the types don't appear the same.  I don't know that Octokit actually has a type anywhere for the status events, so we may just have to build up these mini types.